source/user-manual/capabilities/file-integrity/fim-faq.rst

```diff
@@ -94,3 +95,27 @@ Can I hot-swap monitored directories?
 --------------------------------------
 
 Yes, this can be done for Linux in both agents and manager by setting the monitoring of symbolic links to directories. To set the refresh interval, use option :doc:`syscheck.symlink_scan_interval <../../reference/internal-options>`.
+
+How to edit audit configuration to deal with huge amount of whodata events at the same time.
+----------------------------------------------------------------------------------------
+
+It is possible to lose events when a lot of them are generated in a moment, there is a problem with audit socket and dispatcher that can be fixed by editing two values involved:
+::
+
+  /etc/audisp/audisp.conf  -> disp_qos = ["lossy", "lossless"]
+  /etc/audit/audit.conf    -> q_dephs = [<Numerical value>]
+
+The first one (disp_qos) controls whether you want blocking/lossless or non-blocking/lossy communication between the audit daemon and the dispatcher. There is a 128k buffer between the audit daemon and dispatcher. This is good enogh for most uses. If lossy is chosen, incoming events going to the dispatcher are discarded when this queue is full. (Events are still written to disk if log_format is not nolog.) Otherwise the auditd daemon will wait for the queue to have an empty spot before logging to disk. The risk is that while the daemon is waiting for network IO, an event is not being recorded to disk.
+Valid values are: lossy and lossless.
+Default value is lossy.
+Recommended value is lossless.
+
+The other one (q_dephs) is a numeric value that tells how big to make the internal queue of the audit event dispatcher. A bigger queue lets it handle a flood of events better, but could hold events that are not processed when the daemon is terminated. If you get messages in syslog about events getting dropped, increase this value.
+The default value is 80.
+
+There is another variable that can prevent loss of events, within the syscheck configuration:
```

Comment:
On the Wazuh side, the `rt_delay` variable from the internal FIM configuration can help to prevent the loss of events:
::
    /var/ossec/etc/internal_options.conf  -> rt_delay = [Numerical value]

It sets a delay between real-time alerts in milliseconds. Decrease its value to process who-data events faster.